scripts/benchmarks/benchmark.py

```diff
@@ -374,12 +416,29 @@ def maybe_get_experiment_error_traceback(self):
 
         return None if len(results) == 0 else results
 
+    def get_avg_mem_usage_per_sec_by_device_id(self, min_measurement_in_mib=0):
+        '''
+        This function retrieves the gpu memory logs and returns the average memory consumed per device across the experiment
+        Returns a pd.Series of avg mem usage per sec in MiB for each device id
+        ''' 
+        grouped_indices = pd.read_csv(self.gpu_log_filename, skipinitialspace=True).groupby(by='index')
+        # Calculate the average memory consumption per sec in each device
+        mem_usage_by_device_id = grouped_indices.apply(
+            lambda x: x[GPU_LOG_USED_MEM_COLUMN_NAME].str.replace(GPU_LOG_METRIC_SUFFIX, '').astype(float)
+            )
+        # filter only mem measurements that are above a min MiB value before taking the average mem usage per device
+        mem_usage_by_device_id = mem_usage_by_device_id[mem_usage_by_device_id>min_measurement_in_mib]
```

Comment:
not very sure why this filtering is required.

Reply:
Initial measurement values range from 0-3mb even when the subprocess has started, im filtering out the zero values before i take the mean. If we  use the timestamps difference, it might still have zero values in the series when we take the mean are we okay with that?

Reply:
@achew010 Ok i see what you are getting at. THen I propose the following

y_i above is not the difference between timestamps, it is the difference between the initial GPU mem usage, and the one at the i-th measurement
e.g., if x_0 = 10mb, and x_i = 40mb, then the difference is 30mb

So then if you have a seuqnece x = 10, 10, 20, 30, 40, 40, .... 30, 20, 10
then x_0 = 10 and so

y = 0, 0, 10, 20, 30, 30, ..., 20, 10, 0

Then filter out another within a small range, say 0-3mb

then y_filtered = 10, 20, 30, 30, ... , 20, 10. 

Then report the final as the mean of y_filtered.

Reply:
@fabianlim
I edited the function to subtract the first values from the measurements and return the max value for each device id